packages/core/src/odata-common/request-builder/batch/batch-response-parser.ts

```diff
@@ -122,7 +122,14 @@ export function splitChangeSetResponse(changeSetResponse: string): string[] {
  * @returns A list of sub responses represented as strings.
  */
 export function splitResponse(response: string, boundary: string): string[] {
-  const parts = response.split(`--${boundary}`).map(part => part.trim());
+  const newLineSymbol = detectNewLineSymbol(response);
+  const parts = response.split(`--${boundary}`).map(part => {
+    let trimmedPart = part.trim();
+    if(trimmedPart.includes('204 No Content')){
+      trimmedPart += newLineSymbol;
+    }
+    return trimmedPart;
+  });
```

Comment:
```suggestion
  const parts = response.split(`--${boundary}`).map(part => {
    const trimmedPart = part.trim();
    return trimmedPart.includes('204 No Content')) ? `${trimmedPart}${newLineSymbol}` : trimmedPart;
  });
```

Reply:
With a ternary this might look a little nicer, other than that from my side, lgtm ;)

Reply:
agreed, I should be using ternary more frequent 🤔